src/main/java/com/bookstore/service/cart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,94 @@
+import com.bookstore.dto.item.CartItemRequestDtoWithoutBookId;
+import com.bookstore.dto.item.CartItemResponseDto;
+import com.bookstore.exception.EntityNotFoundException;
+import com.bookstore.mapper.CartItemMapper;
+import com.bookstore.mapper.ShoppingCartMapper;
+import com.bookstore.model.CartItem;
+import com.bookstore.model.ShoppingCart;
+import com.bookstore.model.User;
+import com.bookstore.repository.CartItemRepository;
+import com.bookstore.repository.ShoppingCartRepository;
+import com.bookstore.service.book.BookService;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final BookService bookService;
+
+    @Override
+    public void createCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setId(user.getId());
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartResponseDto findCart() {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        return shoppingCartRepository.findById(user.getId())
+                .map(shoppingCartMapper::toDto)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find shopping cart by id: "
+                        + user.getId()));
+    }
+
+    @Override
+    public ShoppingCartResponseDto saveItem(CartItemRequestDto requestDto) {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        ShoppingCart cart = shoppingCartRepository.findById(user.getId())
+                .orElseThrow(() -> new EntityNotFoundException("Can't find shopping cart by id: "
+                        + user.getId()));
+
+        CartItem cartItem = new CartItem();
+        cartItem.setShoppingCart(cart);
+        cartItem.setBook(bookService.getBookById(requestDto.getBookId()));
+        cartItem.setQuantity(requestDto.getQuantity());
+
+        CartItemResponseDto itemDto = cartItemMapper.toDto(cartItemRepository.save(cartItem));
+        ShoppingCartResponseDto cartDto = shoppingCartMapper.toDto(cart);
+        cartDto.getCartItems().add(itemDto);
+        return cartDto;
+    }
+
+    @Override
+    public ShoppingCartResponseDto update(Long id, CartItemRequestDtoWithoutBookId requestDto) {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        ShoppingCart cart = shoppingCartRepository.findById(user.getId())
```

Comment:
actually, you can extract logic of fetching a cart into separate method, to not duplicate the exception message each time